06-building-a-program-start-to-finish/writing-tests/Output/tests.js

```diff
@@ -0,0 +1,190 @@
+const books = require('../books.json');
+const gotBooks = require('../got_books.json');
+const emptyArray = require('../empty_book.json');
+const invalidContent = require('../invalid_content.json');
+const invalidKey = require('../invalid_book.json');
+
+const index = new InvertedIndex();
+
+describe('Validate File', () => {
+  it('checks to ensure validity of JSON file', () => {
+    expect(index.validateFile(invalidContent)).toBe(false);
+    expect(index.validateFile(invalidKey)).toBe(false);
+    expect(index.validateFile(emptyArray)).toBe(false);
+    expect(index.validateFile(books)).toBe(true);
+    expect(index.validateFile(gotBooks)).toBe(true);
+  });
+});
+
+describe('Tokenize', () => {
+  it('should return "an array of words all lowercase and sorted"', () => {
+    expect(index.tokenize(['Alice', 'In', 'Wonderland', 'What', 'is', 'she', 'Looking', 'for', 'There'])).toEqual(
+            ['alice', 'for', 'in', 'is', 'looking', 'she', 'there', 'what', 'wonderland']
+        );
+  });
+  it('should return "an array of words only and sorted"', () => {
+    expect(index.tokenize(['Ali2ce', 'In', 'Won%de#rland', 'Wh%at', 'i@s', 'she', 'Lo%oking', 'for', 'The$re'])).toEqual(
+            ['alice', 'for', 'in', 'is', 'looking', 'she', 'there', 'what', 'wonderland']
+        );
+  });
+});
+
+describe('createIndex ', () => {
+  const createIndexObject = index.createIndex('books.json', books);
+  it('should return "object" for type of createIndex', () => {
+    expect(typeof createIndexObject).toBe('object');
+  });
+  it('should not be an array', () => {
+    expect(typeof createIndexObject).not.toBe('array');
+  });
+  it('should return true for hasOwnProperty "alice and wonderland"', () => {
+    expect(createIndexObject.hasOwnProperty('alice')).toBe(true);
```

Comment:
Please see [link](https://github.com/airbnb/javascript#objects--prototype-builtins) and refactor this and all other instance on this file